### PR TITLE
Fixed Service annotation method parameters are not in effect

### PR DESCRIPTION
**Pull request by Leishunyu**
## What is the purpose of the change

https://github.com/apache/dubbo/issues/4571

## Brief changelog

The configuration method parameter is not set when the Service annotation is initialized.

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Leishunyu** [comment]
@cvictory can you help me review

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java [hunk 1]
```diff
@@ -447,6 +455,12 @@ private AbstractBeanDefinition buildServiceBeanDefinition(Annotation serviceAnno
 
     }
 
+    private List convertMethodConfigs(Object methodsAnnotation) {
+        if (methodsAnnotation == null){
+            return Collections.EMPTY_LIST;
+        }
+        return MethodConfig.constructMethodConfig((Method[])methodsAnnotation);
+    }
```
Comment (tswstarplanet) [inline dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java]: The code seems a little complexed, I think
```java
Optional.ofNullable(serviceAnnotationAttributes.get("methods")).ifPresent(methods -> builder.addPropertyValue("methods", MethodConfig.constructMethodConfig((Method[]) methods)));
```
one line of code is enough.

Just for reference

**Comment by cvictory** [comment]
@Method was introduced in https://github.com/cvictory/incubator-dubbo/commit/e6957454dc0d90718a9b25ebbbb174e4f3333db2#diff-e95ff65194ba7c202a57c0fea25ff6a9, then is is merge in https://github.com/apache/dubbo/pull/2603 .
And it work at dubbo 2.6 firstly , then it is supported at dubbo 2.7.

I find your pr is not complete, you should also support @method working in @Reference.  And you should modify AnnotationUtils, and remove the code :
```
if (attributeValue.getClass().isAnnotation()) {
                continue;
            }
            if (attributeValue.getClass().isArray() && attributeValue.getClass().getComponentType().isAnnotation()) {
                continue;
            }
```

I have do some test , you can modify and do some test.   If you don't have time , I will work at it instead of you after 3 days.

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java (hunk 1 above)
Comment (tswstarplanet) [reply dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java]: @Leishunyu Do you think it's better to use this to replace the origin implementation ? I think it's simpler and cleaner ?

Comment (Leishunyu) [reply dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java]: I think the code is simple, but the method semantics are not clear enough.According to your way, I think most methods can be rewritten, but this is not necessary.What do you think？

**Pull request merged**